route/src/route_generator_worker.cpp

```diff
@@ -63,6 +63,13 @@ namespace route {
         boost::filesystem::path route_path_object(this->route_file_path_);
         if(boost::filesystem::exists(route_path_object))
         {
+            //after route path object is available to select, worker will able to transit state and provide route selection service
+            if(this->rs_worker_.get_route_state() == RouteStateWorker::RouteState::LOADING) 
+            {
+                this->rs_worker_.on_route_event(RouteStateWorker::RouteEvent::ROUTE_LOADED);
+                publish_route_event(cav_msgs::RouteEvent::ROUTE_LOADED);
```

Comment:
Seems like this publish would be better placed after the for loop below so that the actual loading of the route files has completed.

Reply:
Addressed https://github.com/usdot-fhwa-stol/carma-platform/pull/1086/commits/506a7f9d0dbd44c1ccae950365df060d3e2f8b4a